static/src/js/components/CustomTextWidget/CustomTextWidget.jsx

```diff
@@ -70,8 +70,13 @@ const CustomTextWidget = ({
   // Handle the value changing in the field
   const handleChange = (event) => {
     const { value: newValue } = event.target
+    let valueToSet = newValue
+    // If type is number, remove everything except digits, '.', 'e', 'E', '+' and '-'
+    if (type === 'number') {
+      valueToSet = newValue.replace(/[^\d.eE+-]/g, '')
+    }
 
-    onChange(newValue)
+    onChange(valueToSet)
   }
 
   // Handle the field losing focus
```

Comment:
I know you inherited this, but as we revisit these test files it would probably be a good idea to zip up any remaining tests we still need. For example, line 86 still needs to be fully tested.

Reply:
Fixed the method handleBlur itself, now line 86 is tested.